packages/stryker/src/logging/LogConfigurator.ts

```diff
@@ -0,0 +1,138 @@
+import * as log4js from 'log4js';
+import { LoggerFactory } from 'stryker-api/logging';
+import { LogLevel } from 'stryker-api/core';
+import { minLevel } from './logUtils';
+import LoggingClientContext from './LoggingClientContext';
+import { getFreePort } from '../utils/netUtils';
+
+enum AppenderName {
+  File = 'file',
+  FilteredFile = 'filteredFile',
+  Console = 'console',
+  FilteredConsole = 'filteredConsole',
+  All = 'all',
+  Server = 'server'
+}
+
+const layouts: { color: log4js.PatternLayout, noColor: log4js.PatternLayout } = {
+  color: {
+    type: 'pattern',
+    pattern: '%[%r (%z) %p %c%] %m'
+  },
+  noColor: {
+    type: 'pattern',
+    pattern: '%r (%z) %p %c %m'
+  }
+};
+
+interface AppendersConfiguration {
+  [name: string]: log4js.Appender;
+}
+
+const LOG_FILE_NAME = 'stryker.log';
+export default class LogConfigurator {
+
+  private static createMasterAppenders(consoleLogLevel: LogLevel, fileLogLevel: LogLevel): AppendersConfiguration {
+
+    const multiAppender = { type: require.resolve('./MultiAppender'), appenders: ['filteredConsole'] };
```

Comment:
Why require the type like this? What are you trying to do?

Reply:
That's the log4js way of loading custom appenders. See https://log4js-node.github.io/log4js-node/appenders.html#other-appenders. I'll add the link to the code as well so it's less magical.